### PR TITLE
[Core] Skip also conditions incompatible in TetrahedralMeshOrientationCheck

### LINKED ISSUE #9056: [FluidDynamicsApplication] Quadrilateral and Hexahedra incompatibility
**Description**
I have created a 3D model using hexahedra to mesh the fluid but, at the moment to calculate it, I got errors. It seems like the FluidDynamicsApplication is not ready to use hexahedra since it always checks the tetrahedra orientation in `check_and_prepare_model_process_fluid.py`. The function that checks the orientation crashes if we use another type of element.

Similar issue it is found when I use quadrilateral instead of triangles as conditions. The `navier_stokes_solver_vmsmonolithic.py` has a line where `self.condition_name = "MonolithicWallCondition"` is written hard coded. Kratos does not recognize any element called `MonolithicWallCondition3D4N` so the application crashes as well.

**Scope**
- FluidDynamicsApplication

**Environment**
- OS: Ubuntu 18.04
- Branch: master
- Python 3.7

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #9056.

In this PR basically I added to skip also incompatible conditions in TetrahedralMeshOrientationCheck

Also minor clean up

### CODE REVIEW

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
This process is conceived to work with simplex elements, namely linear tetrahedra and triangles. The problem reported in #9056 comes from the fact that we're calling the process with a wrong mesh. Hence, the fix in my opinion would be to avoid calling the process rather than modifying the process, which behavior is the expected one.

**Comment by rubenzorrilla** [comment]
On top of that, there are many aesthetic changes. Considering that this is a sensitive change in the fluid preprocessing I would leave such aesthetics for a future PR.

**Comment by loumalouomega** [comment]
> On top of that, there are many aesthetic changes. Considering that this is a sensitive change in the fluid preprocessing I would leave such aesthetics for a future PR.

The real change is the check of line and triangle

File: kratos/processes/tetrahedral_mesh_orientation_check.cpp [hunk 1]
```diff
@@ -75,26 +75,31 @@ void TetrahedralMeshOrientationCheck::Execute()
...
-        DenseVector<int> ids(geom.size());
+        GeometryType& r_geometry = it_cond->GetGeometry();
 
-        for(IndexType i=0; i<ids.size(); i++) {
-            geom[i].Set(BOUNDARY,true);
-            ids[i] = geom[i].Id();
-        }
+        const GeometryData::KratosGeometryType geometry_type = r_geometry.GetGeometryType();
+
+        if (geometry_type == GeometryData::Kratos_Triangle3D3  || geometry_type == GeometryData::Kratos_Line2D2) {
```
Comment (loumalouomega) [inline kratos/processes/tetrahedral_mesh_orientation_check.cpp:84]: This is the change (you check elements, but not conditions)

**Comment by loumalouomega** [comment]
> On top of that, there are many aesthetic changes. Considering that this is a sensitive change in the fluid preprocessing I would leave such aesthetics for a future PR.

The total number of changes is still small, have you check it?

**Comment by rubenzorrilla** [comment]
> > On top of that, there are many aesthetic changes. Considering that this is a sensitive change in the fluid preprocessing I would leave such aesthetics for a future PR.
> 
> The total number of changes is still small, have you check it?

I did. But I insist, this is a sensitive preprocessing change so aesthetics must be removed from this PR, which is a behavior change.

**Comment by loumalouomega** [comment]
> > > On top of that, there are many aesthetic changes. Considering that this is a sensitive change in the fluid preprocessing I would leave such aesthetics for a future PR.
> > 
> > 
> > The total number of changes is still small, have you check it?
> 
> I did. But I insist, this is a sensitive preprocessing change so aesthetics must be removed from this PR, which is a behavior change.

Now, just adding the change

**Comment by loumalouomega** [comment]
> > > > On top of that, there are many aesthetic changes. Considering that this is a sensitive change in the fluid preprocessing I would leave such aesthetics for a future PR.
> > > 
> > > 
> > > The total number of changes is still small, have you check it?
> > 
> > 
> > I did. But I insist, this is a sensitive preprocessing change so aesthetics must be removed from this PR, which is a behavior change.
> 
> Now, just adding the change

It is a very simple check, #9056 will be fixed as the problematic conditions will be skipped

File: kratos/processes/tetrahedral_mesh_orientation_check.cpp (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/processes/tetrahedral_mesh_orientation_check.cpp:84]: OK. going through the rest of the file I realized that we're checking that the elements are simplex but we're not doing so to ensure that the condition types are the corresponding ones.

**Review by rubenzorrilla — APPROVED** [review]
As we're doing the same for the elements we can accordingly check the conditions.

In any case, although this removes the error reported by @jgonzalezusua, it does not completely fix the problem as we skip the parents search and normal calculation, something that might eventually needed. Having said this, we need find a way to extend this to hexas/quads.

**Comment by loumalouomega** [comment]
> As we're doing the same for the elements we can accordingly check the conditions.
> 
> In any case, although this removes the error reported by @jgonzalezusua, it does not completely fix the problem as we skip the parents search and normal calculation, something that might eventually needed. Having said this, we need find a way to extend this to hexas/quads.

This was supposed to be a fast PR, to fix an issue notified by a user ASAP, so he can continue with it workflow with minimal impact. Of course, it does not fix the problem, which can be solved with not much effort, but with a true refactor of the process (including name, which is misleading)

**Comment by rubenzorrilla** [comment]
I wonder if we should add a warning in both condition/element checks to inform the user that neighbors and normals are not to be computed.

**Comment by jgonzalezusua** [comment]
@rubenzorrilla, @loumalouomega  thanks for the attention received!!!

**Pull request merged**